### ISSUE TITLE
Add support for a continuous time AR(1) analogue to cor_ar()

### ISSUE THREAD

**Issue by gavinsimpson** [paul-buerkner/brms#741]
The discrete time AR(1) correlation structure can be generalised to the continuous time setting.

A continuous time AR(1) or CAR(1) correlation structure can be defined as

h(s, &straightphi;) = &straightphi;^s, s &geq; 0, &straightphi; &geq; 0

where s is a non-negative real. In contrast to the AR(1), the correlation parameter &straightphi; is constrained to be non-negative. The CAR(1) correlation function is a univariate special case of the exponential spatial correlation function.

The *nlme* package has an implementation of this correlation function in its `nlme::corCAR1()`.

A `cor_car1()` function in brms mirroring `cor_ar()` but for the CAR(1) correlation would be very useful for modelling longitudinal data for example in settings where the observations are not regularly spaced in time.

There is a small amount of [discussion](https://discourse.mc-stan.org/t/autocorrelation-for-unevenly-spaced-time-series/10001/6) of this on the Stan Discourse site.

**Comment by paul-buerkner** [comment]
Thanks for opening this issue! Can we think of another name as cor_car1 though as it is too close to the cor_car function for (spatial) conditionally autoregessive models and thus will confuse users.

**Comment by gavinsimpson** [comment]
Would `cor_ctar1()` be sufficiently different or do we need to think further from the box?

**Comment by paul-buerkner** [comment]
Sounds reasonable. Perhaps we should go for `cor_ctar` and make the `1` an argument in case we ever want to go beyond processes of order 1.

**Comment by gavinsimpson** [comment]
`cor_ctar()` sounds good to me. :+1:

**Comment by GitTFJ** [comment]
Has the `cor_ctar` function been added yet? I cant seem to find any other (except this one) reference to it. If not, are there any plans to add this continuous temporal term?

**Comment by paul-buerkner** [comment]
It has not been added yet unfortunately. There are plans as per this issue but no specific timeline.

**Comment by jan-joh** [comment]
I am working on this. I have already implemented the calculation of the time differences between samples to include in the standata. Unfortunately, my mathematical understanding of the process is very limited. Could anybody with a deeper understanding (maybe @paul-buerkner ) confirm if this change in the code would be sufficient, or if there are other terms that would need to be added?

### Standard AR
```stan
mu += Intercept + err;
// include ARMA terms
for (n in 1:N) {
  for (i in 1:J_lag[n]) {
    Err[n + 1, i] = err[n + 1 - i];
  }
  mu[n] += Err[n, 1:Kar] * ar;
}
```

### ctAR(p=1)
(Note that I have restricted the helper to p=1 and q=0 for now)

```stan
  mu += Intercept + err;
  // include ARMA terms
  for (n in 1:N) {
    for (i in 1:J_lag[n]) {
      Err[n + 1, i] = err[n + 1 - i];
    }
    mu[n] += Err[n, 1:Kar] * pow(ar, J_tdelta[n]);
  }
```

So basically, I would just substitute Err * ar -> Err * ar^<time_difference>.

Additionally constraining AR to `vector<lower=0,upper=1>[Kar] ar` of course.

**Comment by paul-buerkner** [comment]
Very cool, thank you! Yes, this is how I would implement it too. Minor note, I am using J_* for indices but here it is the actual time difference. So perhaps delta_t or something that fits to the rest of the names for ARMA models could make sense.
